### PR TITLE
Added an setting to display the actual odds with true hit enabled

### PR DESCRIPTION
**Pull request by wellme**
Addresses #110 basically.
Pretty much self-explanatory.

### CODE REVIEW

File: src/net/fe/fightStage/CombatCalculator.java [hunk 1]
```diff
@@ -415,8 +426,66 @@ public static int hitRate(Unit a, Unit d){
...
+		public String formattedCrit() {
+			if(crit < 0)
+				return " --";
+			if(FEResources.getActualOdds())
+				return String.format("%3d", FEMultiplayer.getSession().getCritRNG().actualSuccessRate(crit));
+			return String.format("%3d", crit);
+		}
+		
+		public boolean canAttack() {
+			return crit < 0;
```
Comment (rayrobdod) [inline src/net/fe/fightStage/CombatCalculator.java]: Other way around - can attack if `crit >= 0`. This commit disabled the 'x2' preview thing only if the unit is capable of counter-attacking, when we wanted the inverse.

**Comment by eliatlarge** [comment]
I've done the same thing multiple times on this project, glad you caught it :P

**Comment by SapphireTactician** [comment]
@rayrobdod Thank you ever so much for reviewing the code.

Also, I apologize for being so hands-off with everything as of late.

I just finished putting this branch through its paces and can confirm both that the feature works as intended and that the issue Ray found had been fixed.

@wellme Before we merge, I'd like to voice a few concerns:

* Why is the default `false` when this is a community requested feature?
* As a setting, I feel this should be server side, not client side, in case the host wants to enforce the old True Hit display be it for nostalgia or whatever unimaginable purpose.

**Comment by wellme** [comment]
It defaults to false because that's the behavior of FE games. ~~Also because I fundamentally disagree with applications changing their behavior after updates.~~

Regarding whether it's server-side or client-side, I really don't think it matters. Like you said, it would only be useful if the server owner wanted to restrict the game to the old true hit display, which is pointless because there are true hit calculators that will do the conversion for you. I think this setting is really just a matter of preference (even though almost everyone will prefer to see the actual numbers), hence why it's client sided.

This is my reasoning for the decisions that I took while coding the feature. If you think it would be better otherwise, I'll implement your changes. You are probably in a way better spot to make decisions regarding how the game should be than me. I mean, I don't even play the game myself.

**Comment by SapphireTactician** [comment]
@wellme @rayrobdod I might've spoken too soon about everything being fine aside from the two concerns I stated in my previous comment. Noticing commit 67fa311 I decided to test again under the following conditions:

* Hit RNG: `True Hit`
* Display Actual Odds: `true`

And I noticed a rather huge discrepancy between when using and not using the Vegas modifier (*Halves Hit but doubles Crit*).

**Actual Odds without Vegas:**
2018-04-03--195239_1600x900

*****

**Actual Odds WITH Vegas:**
2018-04-03--195530_1600x900

*****

Somewhere, somehow is some really bad math. If Gamble is halving the false True Hit odds (*which is 37 in this same test case*) then according to the chart [here](https://serenesforest.net/general/true-hit/) the actual odds should be around 7%. If Gamble is halving the actual odds, as in after true hit is already applied, then it should be much higher. So my question is why are the actual odds with Gamble displaying as 0???

**Comment by SapphireTactician** [comment]
>Regarding whether it's server-side or client-side, I really don't think it matters. Like you said, it would only be useful if the server owner wanted to restrict the game to the old true hit display, which is pointless because there are true hit calculators that will do the conversion for you.

@wellme Good point. We'll leave that as is then, maybe even leave the default as-is too, but if you can then, make the setting reliable accessible to the end user via the `Options` menu on the client.

**Comment by wellme** [comment]
Gamble reduces the hit _stat_ by 50%, not the actual hit rate in a given battle. This is how it was implemented originally and so I'm tempted to believe this is actually intended behavior. I'll test some more to see if the math works out.

I'm not sure what gamble actually does in FE, though. We might want to change how gamble works anyway considering how it pretty much trashes all accuracy to nothingness.

**Comment by SapphireTactician** [comment]
>Gamble reduces the hit stat by 50%, not the actual hit rate in a given battle.

This is incorrect, the Gamble skill halves the hit rate.
*Source: http://fireemblem.wikia.com/wiki/Gamble*

Furthermore, speaking of Hit Rate, IIRC, the Hit Rate uses the following formula (*excluding elements we don't have yet such as Weapon Rank, Support, and Tactician Bonus*): `Hit Rate = (Skl ×2) + (Lck ×0.5) + (Weapon Hit)` so if it was meant to be Hit stat, which it isn't, then are you implying that it currently halves the weapon's hit, or the user's Skill?

**Comment by wellme** [comment]
What I meant was that this is how it's implemented in the game _now_. My point was that this issue is separate from this PR.

What gamble does currently (in other branches too) is divide the user hit rate in half, in such a way that the _actual_ hit rate formula is `Hit Rate = ((Skl ×2) + (Lck ×0.5) + (Weapon Hit)) / 2`.

Example (ignoring all other modifiers, such as terrain, triangle, non-gamble skills, etc.):

Lyn has 150 base hit
Eirika has 50 base avoid

Without gamble: 150 - 50 = 100 hit rate
With gamble: 150/2 - 50 = ***25*** hit rate

**Comment by SapphireTactician** [comment]
I'm not blaming the PR, but the goal of this PR is to bring transparency to the hit rate, and if part of that involves factoring in Gamble, then we might as well fix Gamble along with it imo since Gamble is not behaving as faithfully.

I understand that you're referring to the way it is now, but that doesn't make it any more correct. You are correct though that I forgot to factor in things like Avoid and the WTA. However, according to this source ( http://fireemblem.wikia.com/wiki/Hit_Rate ) those things also count towards the hit rate affected by Gamble, according to the link I shared earlier about Gamble.

That being said, I know it's no fault of yours, but can you please fix Gamble while you're at it. I don't see much of a reason for it to be a separate PR, especially when I need to make sure it works correctly with this one.

**Comment by wellme** [comment]
Well alright. It'll take some time to implement because the current structure of the combat calculator isn't all that flexible.

Regarding the options menu: the unfortunate fact is that I have 0 creative ability and the current interface forces me to put an icon for the option. I'll try my best, but I wouldn't expect a miracle.

**Comment by SapphireTactician** [comment]
@wellme Nevermind, you're right. I totally misread the wiki.... today is not my day at all... Although it probably would work much better the way I misread it....

**Comment by wellme** [comment]
Honestly, I have no idea what would be best, so I'll default to the first choice and open an issue later.

**Comment by SapphireTactician** [comment]
>Regarding the options menu: the unfortunate fact is that I have 0 creative ability and the current interface forces me to put an icon for the option. I'll try my best, but I wouldn't expect a miracle.

@wellme Something simple like a die / dice would do imo.

**Comment by wellme** [comment]
No it's fine, I still need to come up with an icon for the option menu. Either that or ask for someone to do it for me. I should at least try asking the discord server before wasting time creating a mediocre icon.

**Comment by rayrobdod** [comment]
I feel like there has to be something better than a die to represent "change how chances are reported", but I haven't figured out what that might be yet.

**Comment by rayrobdod** [comment]
`optionIcon.png` has what appear to be lossy compression artifacts.

**Comment by wellme** [comment]
I don't see it. Is it still present with the new version? (Had to update it because of the fog color option introduced in #154)

**Comment by rayrobdod** [comment]
Admittedly, I can barely see it even when zoomed in, but the fill tool, among others, can tell that there are too many shades of grey and white here.

File: src/net/fe/OptionsStage.java [hunk 2]
```diff
@@ -81,6 +81,10 @@ public void execute() {
 			new OptionGroupEntity(140,
 				new OptionGroup("FOG COLOR", java.util.Arrays.asList("DAY", "NIGHT"), java.util.Arrays.asList("Makes fog-of-war appear white, washing-out the terrain", "Makes fog-of-war appear black, dimming the terrain")),
 				FEResources.getFogColor().ordinal()
+			),
+			new OptionGroupEntity(170,
+				new OptionGroup("ACTUAL ODDS", java.util.Arrays.asList("TRUE", "FALSE"), java.util.Arrays.asList("Display the actual odds of a hit/crit happening ", "Display the numerical result of the hit/crit calculation")),
+				FEResources.getActualOdds() ? 1 : 0
```
Comment (rayrobdod) [inline src/net/fe/OptionsStage.java]: The ordering in the list-of-values and ordering in the initially-selected-value-index needs to match; here the list of values has true first and false second, while the initially selected value has false first and true second.

**Pull request merged**